### ISSUE TITLE
BuildScanPlugin.kt's system() call does not return if process produces lots of output on Windows

### ISSUE THREAD

**Issue by lptr** [gradle/gradle#7016]
[This code](https://github.com/gradle/gradle/blob/44be59e469df8d4e055c19f223270566824bd833/buildSrc/subprojects/profiling/src/main/kotlin/org/gradle/gradlebuild/profiling/buildscan/BuildScanPlugin.kt#L219-L227) fails to return if the executed command produces lots of output on Windows:

```kotlin
private
fun Project.system(vararg args: String): String =
    ProcessBuilder(args.toList())
        .directory(rootDir)
        .start()
        .run {
            assert(waitFor() == 0)
            inputStream.bufferedReader().use { it.readText().trim() }
        }
```

The problem is Windows won't let the child process exit until both output and error streams are fully read. There is some buffering that allows a small amount of output to be handled, but running something like `git status` can produce a lot of output that makes the child process stop waiting for its output to be unblocked.

The solution should be to read the output of the process _before_ waiting for it to finish.

Note that the error output should produce similar problems, especially since error output is not buffered, so it will block as soon as any error output is produced by the child process.

**Comment by oehme** [comment]
We should just use `project.exec` here, that has all the proper handling we need.

**Comment by ldaley** [comment]
This title is misleading. The problem is not in _the_ build scan plugin.

**Comment by lptr** [comment]
I renamed it.

**Comment by lptr** [comment]
Fixed via 91a0aac8bd9f.
